packages/admin-ui-core/src/styles.ts

```diff
@@ -0,0 +1,129 @@
+import type { Emotion } from '@emotion/css/create-instance'
+import type { CSSObject as EmotionCSSObject } from '@emotion/css'
+import { isFunction } from '@vtex/admin-ui-util'
+
+import { alias } from './aliases'
+import { resolveRule } from './rules'
+import { canSplit, split } from './splits'
+import { createTransform } from './transforms'
+import type { StyleObject, StyleProp, ThemeDerivedStyles } from './types'
+import { theme as defaultTheme } from './theme'
+
+/**
+ * Available system breakpoints
+ */
+const breakpoints = ['40em', '48em', '64em', '75em']
+
+const TOKEN_PREFIX = '$'
+
+/**
+ * Available media queries mapped from the breakpoints
+ */
+const mediaQueries = [
+  null,
+  ...breakpoints.map((n) => `@media screen and (min-width: ${n})`),
+]
+
+/**
+ * This function allows the responsive arrays
+ * @example
+ * csx({
+ *  width: [20, 30, 40]
+ * })
+ */
+function resolveResponsiveArray(
+  styles: Exclude<StyleProp, ThemeDerivedStyles>
+) {
+  const next: Exclude<StyleProp, ThemeDerivedStyles> = {}
+
+  for (const k in styles) {
+    const key = k as keyof typeof styles
+    const value = styles[key]
+
+    if (value == null) continue
```

Comment:
**Suggestion:** If you to check if value is `null` or not it could be better use `value === null`, because `undefined == null` is `true`, but if you also want to consider value as undefined, maybe  `!!value` could it be a better option.

Reply:
Tried, but. The logic changes. So, I'm keeping that. The truth about this function is that we must re-write it. I know the logic is not perfect right now. But, I prefer keeping that because I'm refactoring. In another PR I can upgrade the logic around it. What do you think?

Reply:
I agree, make sense ;)